python/lsst/pipe/tasks/make_direct_warp.py

```diff
@@ -19,956 +19,19 @@
-    )
-
-    def __init__(self, *, config=None):
-        super().__init__(config=config)
-        if not config:
-            return
-
-        if not config.doRevertOldBackground:
-            del self.background_revert_list
-        if not config.doApplyNewBackground:
-            del self.background_apply_list
-
-        if not config.doWarpMaskedFraction:
-            del self.masked_fraction_warp
-
-        # Dynamically set output connections for noise images, depending on the
-        # number of noise realization specified in the config.
-        for n in range(config.numberOfNoiseRealizations):
-            noise_warp = Output(
-                doc=f"Output direct warped noise exposure ({n})",
-                name=f"{config.connections.coaddName}Coadd_directWarp_noise{n}",
-                # Store it as a MaskedImage to preserve the variance plane.
-                storageClass="MaskedImageF",
-                dimensions=("tract", "patch", "skymap", "instrument", "visit"),
-            )
-            setattr(self, f"noise_warp{n}", noise_warp)
-
-
-class MakeDirectWarpConfig(
-    PipelineTaskConfig,
-    pipelineConnections=MakeDirectWarpConnections,
-):
-    """Configuration for the MakeDirectWarpTask.
-
-    The config fields are as similar as possible to the corresponding fields in
-    MakeWarpConfig.
-
-    Notes
-    -----
-    The config fields are in camelCase to match the fields in the earlier
-    version of the makeWarp task as closely as possible.
-    """
-
-    MAX_NUMBER_OF_NOISE_REALIZATIONS = 3
-    """
-    numberOfNoiseRealizations is defined as a RangeField to prevent from
-    making multiple output connections and blowing up the memory usage by
-    accident. An upper bound of 3 is based on the best guess of the maximum
-    number of noise realizations that will be used for metadetection.
-    """
-
-    numberOfNoiseRealizations = RangeField[int](
-        doc="Number of noise realizations to simulate and persist.",
-        default=0,
-        min=0,
-        max=MAX_NUMBER_OF_NOISE_REALIZATIONS,
-        inclusiveMax=True,
-    )
-    seedOffset = Field[int](
-        doc="Offset to the seed used for the noise realization. This can be "
-            "used to create a different noise realization if the default ones "
-            "are catastrophic, or for testing sensitivity to the noise.",
-        default=0,
-    )
-    useMedianVariance = Field[bool](
-        doc="Use the median of variance plane in the input calexp to generate "
-            "noise realizations? If False, per-pixel variance will be used.",
-        default=True,
-    )
-    doRevertOldBackground = Field[bool](
-        doc="Revert the old backgrounds from the `background_revert_list` "
-            "connection?",
-        default=False,
+# TODO: Remove this entire file in DM-47521.
+
+import warnings
+
+try:
+    from lsst.drp.tasks.make_direct_warp import *  # noqa: F401, F403
```

Comment:
I assume this wildcard import is safe. Same in `python/lsst/pipe/tasks/make_psf_matched_warp.py`.

Reply:
Well, given that it is meant to mimic the one in `drp_tasks`, it is. But I'll make it limited to what was originally present at the time of this transfer.